### PR TITLE
added search ability to Activity Log AGPUSH-1390

### PR DESCRIPTION
**Pull request by lfryc**
https://issues.jboss.org/browse/AGPUSH-1390

What:

* we want to search for messages by their content (in fact we can search for anything in raw JSON message)

Testing:

* send more than 10 testing notifications so that you have >=2 pages in the Activity Log
* try search for some alert strings in the table header on Activity Log page

### CODE REVIEW

File: jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/metrics/PushMetricsEndpoint.java [hunk 1]
```diff
@@ -16,16 +16,20 @@
...
-import org.jboss.aerogear.unifiedpush.service.metrics.PushMessageMetricsService;
+import static org.jboss.aerogear.unifiedpush.rest.util.HttpRequestUtil.extractSortingQueryParamValue;
 
 import javax.inject.Inject;
-import javax.ws.rs.*;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
```
Comment (matzew) [inline jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/metrics/PushMetricsEndpoint.java]: PLUS PLUS

File: model/api/src/main/java/org/jboss/aerogear/unifiedpush/dao/PushMessageInformationDao.java [hunk 2]
```diff
@@ -70,7 +70,7 @@
      *
      * @return list of push message info objects
      */
-    PageResult<PushMessageInformation> findAllForPushApplication(String pushApplicationId, boolean ascending, Integer page, Integer pageSize);
+    PageResult<PushMessageInformation> findAllForPushApplication(String pushApplicationId, String search, boolean ascending, Integer page, Integer pageSize);
```
Comment (matzew) [inline model/api/src/main/java/org/jboss/aerogear/unifiedpush/dao/PushMessageInformationDao.java]: JavaDoc needs `@param` for search as well (yay!)

**Comment by matzew** [comment]
@lfryc see here: https://github.com/lfryc/aerogear-unifiedpush-server/pull/9

**Comment by sebastienblanc** [comment]
:+1:

**Comment by matzew** [comment]
landed on master 27aa85b47cdf93a67993692b923bc7ab6ede00ab

**Pull request closed without merging**